### DISCUSSION TITLE
Walkthrough or code for enabling secret auto-rotation

### DISCUSSION THREAD

**Discussion by jbla9028** [Azure/secrets-store-csi-driver-provider-azure#548] [Q&A] [answered]
Hello, I'm trying to figure out how to enable auto-rotation and where this flag should go + syntax. Does someone have an example?

https://secrets-store-csi-driver.sigs.k8s.io/topics/secret-auto-rotation.html#auto-rotation-of-mounted-contents-and-synced-kubernetes-secrets

To enable auto rotation, enable the **--enable-secret-rotation**  feature gate for the secrets-store container in the Secrets Store CSI Driver pods. The rotation poll interval can be configured using --rotation-poll-interval. The default rotation poll interval is 2m. If using helm to install the driver, set enableSecretRotation: true and configure the rotation poll interval by setting rotationPollInterval. The rotation poll interval can be tuned based on how frequently the mounted contents for all pods and Kubernetes secrets need to be resynced to the latest.

I am a bit new to k8s but got the provider working and want to test this process but the docs weren't super clear on this step me :)

**Comment by nilekhc** [comment, marked as answer]
Hello @jbla9028 ,
This flag is set in Secret Store CSI driver. Checkout this deployment configuration from [driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/f430a74f5bbf0892651d0241dd744e94d93e50c5/deploy/secrets-store-csi-driver.yaml#L51).

If you have installed Driver and Provider via helm, then you need to set that flag while installing the chart.

for eg.
```
helm install csi-secrets-store-provider-azure/csi-secrets-store-provider-azure --generate-name --set secrets-store-csi-driver.enableSecretRotation=true
```

**Reply by jbla9028** [reply]
Thank you.
